### PR TITLE
Generic parsing

### PR DESCRIPTION
**Pull request by llmlks**

### CODE REVIEW

**Comment by henrhaat** [comment]
The parser works as intended, when the data is in the right format. I also tested two different 'wrong' types of data called generic_test1.txt and generic_test2.txt that I put in Trello.

generic_test1.txt: The year in the first row is 19893 and the app says: "File generic_test1.txt uploaded successfully, 0 rows. Please select a tool from the menu to inspect the data."

generic_test2.txt: The first row is missing an ID: "File generic_test2.txt uploaded successfully, 10 rows. Please select a tool from the menu to inspect the data." The views give wrong graphs and the data view throws an error.

The first is not that big a deal since it says that 0 rows were uploaded. For the second one it could, if possible, give the same error as for other wrong data types, if there is time and it's a quick fix.

These are just my observations and provided the user has the data correctly, this works.

File: instructions.md [hunk 1]
```diff
@@ -11,6 +11,14 @@ To get started, please upload the catalog of your choice by clicking the upload
...
 Once the data has been uploaded, the available features include a map view, a scatter plot view, and a clustering view. Each of the features includes a set of configurations to control them. With the exception of the clustering view, changes in the settings only take effect after submitting them by clicking on the "Apply" button on each page.
 
+**Generic parser**: You can also use another catalog type, not based on an existing catalog. The file needs to be uploaded as a .txt file, and the file structure should be as follows
+
+The overall format is
+```
+id,year,month,day,hour,minute,second,depth,magnitude,latitude,longitude
+```
+No header should be added to the file. Each event is on its own row, and the values are separated by commas. The ID field does not need to be numeric, but all of the other fields only accept numbers.
+
```
Comment (niklasmaki) [inline instructions.md]: What is the latitude/longitude format? Can the file contain missing values?

**Review by niklasmaki — COMMENTED** [review]
I've been thinking about Tommi's comments that .txt extension is too general and that we shouldn't use it for the FENCAT catalog. With the same logic the .csv extension is also too general, and should not be used for the Otaniemi catalog. Also, it's kind of strange that the file format of the general parser is essentially CSV but it can't have the .csv extension.

So I would suggest the following:
1. Use both .csv and .txt as generic parser extensions
2. Invent a new extension for the Otaniemi data (or ask Tommi)
3. Add the extensions to the instructions

I think the extensions should be very clearly visible in the instructions (maybe right after each dataset link?) especially if we change the Otaniemi extension to a non-default one. Or of course we could add manual selection of parsers but I think it's too late for that.

Or what do you think?

**Review by niklasmaki — CHANGES_REQUESTED** [review]
I think this looks good. I tested the example file in Trello and the generic parser works with both .txt and .csv extensions.

However, maybe the instructions could be even more specific about the lat/long format? There are multiple formats how latitudes and longitudes can be presented, see e.g. [here](http://www.geomidpoint.com/latlon.html).

**Pull request merged**